x/gov/handler.go

```diff
@@ -311,20 +321,40 @@ func (h CreateProposalHandler) Deliver(ctx weave.Context, db weave.KVStore, tx w
 		ElectionRuleRef: orm.VersionedIDRef{ID: msg.ElectionRuleID, Version: rule.Version},
 		ElectorateRef:   orm.VersionedIDRef{ID: rule.ElectorateID, Version: electorate.Version},
 		VotingStartTime: msg.StartTime,
-		VotingEndTime:   msg.StartTime.Add(rule.VotingPeriod.Duration()),
+		VotingEndTime:   votingEnd,
 		SubmissionTime:  weave.AsUnixTime(blockTime),
 		Author:          msg.Author,
 		VoteState:       NewTallyResult(rule.Quorum, rule.Threshold, electorate.TotalElectorateWeight),
 		Status:          Proposal_Submitted,
 		Result:          Proposal_Undefined,
 		ExecutorResult:  Proposal_NotRun,
+		TallyTaskID:     nil, // Chicken-egg problem. Create without and update later.
 	}
 
 	obj, err := h.propBucket.Create(db, proposal)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to persist proposal")
 	}
 
+	tallyMsg := &TallyMsg{
+		Metadata:   &weave.Metadata{Schema: 1},
+		ProposalID: obj.Key(),
+	}
+	// Add two seconds for the margin, because tally logic is using one second margin.
+	runAt := votingEnd.Time().Add(2 * time.Second)
+	// Tally message requires no authentication.
```

Comment:
I like all these lines.
Pretty simple to schedule a task for the future and even store a reference to the id.
Well done